src/BenchmarkDotNet.Annotations/Attributes/GenericTypeArgumentsAttribute.cs

```diff
@@ -11,6 +12,6 @@ public class GenericTypeArgumentsAttribute : Attribute
         // CLS-Compliant Code requires a constructor without an array in the argument list
         [PublicAPI] public GenericTypeArgumentsAttribute() => GenericTypeArguments = new Type[0];
 
-        public GenericTypeArgumentsAttribute(params Type[] genericTypeArguments) => GenericTypeArguments = genericTypeArguments;
+        public GenericTypeArgumentsAttribute([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] params Type[] genericTypeArguments) => GenericTypeArguments = genericTypeArguments;
```

Comment:
Unfortunately placing the DAMAttribute on an array is a no-op. Arrays are difficult to model in a static analysis.

This should be generating a warning along the lines of `Trim analysis warning IL2098: GenericTypeArgumentsAttribute..ctor(Type[]): Parameter '#1' of method 'GenericTypeArgumentsAttribute..ctor(Type[])' has 'DynamicallyAccessedMembersAttribute', but that attribute can only be applied to parameters of type 'System.Type' or 'System.String'.`

Reply:
this actually broke building `dotnet/performance` with `dotnet/runtime`. When building a wasm project, I get - https://gist.github.com/radical/b75b5e4063d97be0adac729b611990ea . Can we revert this change? @adamsitnik

Reply:
If I remove that attribute from the param, then it works fine.

Reply:
To be clear, this breaks *when* I update `bdn` reference in `dotnet/performance` to `0.13.1.1740`.

Reply:
The attribute is a no-op - it should be removed because even in the absence of the illink bug, it generates a warning. NativeAOT doesn't have the linker bug and that's why it works there. I filed https://github.com/dotnet/linker/issues/2714 on the linker bug - that one should also be fixed independently of this removal.

Reply:
I've sent https://github.com/dotnet/BenchmarkDotNet/pull/1973 to address that. @radical is there any chance you could test it to see if it actually helps?

Reply:
I can build with that change 👍